src/utils.jl

```diff
@@ -120,6 +120,81 @@ function broadcast_edges(g::GNNGraph, x)
     return gather(x, gi)
 end
 
+function _sort_col(matrix::AbstractArray; rev::Bool = true, sortby::Int = 1)
+    index = sortperm(view(matrix, sortby, :); rev)
+    return matrix[:, index], index
+end
+
+function _topk_matrix(matrix::AbstractArray, k::Int; rev::Bool = true, sortby::Union{Nothing, Int} = nothing)
+    if sortby === nothing
+        sorted_matrix = sort(matrix, dims = 2; rev)[:, 1:k]
+        vector_indices = map(x -> sortperm(x; rev), eachrow(matrix))
```

Comment:
instead of sorting the whole matrix, it would be more efficient to use `partialsortperm`. I'm not sure is supported by CUDA.jl though